GNSSLogger/app/src/main/java/com/google/android/apps/location/gps/gnsslogger/SensorFusionContainer.java

```diff
@@ -251,6 +281,24 @@ public void registerLocation() {
     logRegistration("LocationUpdates", isGpsProviderEnabled);
   }
 
+  public void unregisterLocation() {
+    mLocationManager.removeUpdates(mLocationListener);
+  }
+
+  public void registerFusedLocation() {
+    LocationRequest locationRequest = new LocationRequest();
+    locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+    locationRequest.setInterval(1000);
+    locationRequest.setFastestInterval(100);
+    LocationServices.FusedLocationApi.requestLocationUpdates(mGoogleApiClient, locationRequest, mFusedLocationListener);
```

Comment:
In this class you've already got mLocationProvider as a field, and in another PR you're adding mSensorManager as a field. For symmetry, make the the FusedLocationClient a field as well. Instantiate it in the class constructor, like `mFusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(context)`.

Reply:
I agree that this suggestion makes the class implementation more consistent. Unfortunately it looks like the FusedLocationProviderClient has bugs before Google Play services version 11.6.0:

https://developers.google.com/android/reference/com/google/android/gms/location/FusedLocationProviderClient

I tried changing the Gradle configuration to upgrade play-services-location to version 11.6.0, but I wasn't able to build successfully until I upgraded both play-services-location and play-services-maps to version 12.0.0. 

It looks like both the FusedLocationProviderApi and ActivityRecognitionApi were deprecated starting in version 12, so to implement your suggestion I believe I'd have to convert both to their respective Client versions. FusedLocationProviderClient has different method signatures for requestLocationUpdates() than FusedLocationProviderApi, so if I'm not missing something (which I definitely could be), this will significantly increase the scope of this change/PR.

Would you like me to go ahead and make those changes? If so, should I add more commits to this PR, or create a new one? There are several other settings in the Gradle configuration that could use upgrading, so maybe this is a worthwhile effort, but I was hesitant to go that far initially without gauging interest from the maintainers.

Reply:
Please feel free to make that upgrade in the future, but you're right, we can keep that outside the scope of this PR. If you'd like to push that change in the future, I'd be happy to review it.